src/App.js

```diff
@@ -1,24 +1,100 @@
-import logo from './logo.svg';
+
+import { useState } from 'react';
+import contactsArr from "./contacts.json";
+
 import './App.css';
 
+
+
 function App() {
+  let fiveContacts = contactsArr.slice(0,5);
+
+  const [contacts, setContacts] = useState(fiveContacts);
+
+  const addRandomContact = () => {
+  if(!contactsArr.length){ return false;}
+
+    const randomContact = Math.floor(Math.random() * (contactsArr.length));
+    const newRandomContact = [...contacts,contactsArr[randomContact]]
+   contactsArr.splice(randomContact,1);
+    setContacts(newRandomContact);
+  }
+
```

Comment:
also hier solltet ihr Kontakte aus contactArr nehmen, aber nur Kontakte, die noch nicht dran waren.
Also mit der filter Methode hier werden in newContactsList, nur Kontakte gespeichert, die noch nicht im State sind. Also anfangs die "fiveContacts" ausgeschlossen und später dann alle die wir "random" hinzufügen. Sodass wir keine doppelt haben. Dann wählen wir einen random Contact aus der Liste. Letzter Schritt ist den neuen random Contact mit den vorherigen Kontakten zusammen zu fügen und das zu returnen. Hier wählen wir die "option 2" wie Luis sie nannte mit callback und return für setState, weil der neue State auf den vorherihen State aufbaut.

 const addRandomContact = ()=>{
    setContacts((prevContacts)=>{
      const newContactsList = contactsArr.filter((contact)=> {
        return !prevContacts.includes(contact);
      });
      const getRandomContact = newContactsList[Math.floor(Math.random()* newContactsList.length)];
      const newConctactSelection = [...prevContacts,getRandomContact]
      return newConctactSelection;
    })
  };